enterprise/models.py

```diff
@@ -272,6 +272,37 @@ def user_email(self):
             return self.user.email
         return None
 
+    @property
+    def entitlements(self):
```

Comment:
Okay - so is this API designed to be used by the ecommerce stack to determine which entitlements a particular user is eligible to receive?

If so, then I think gating on course-specific consent is a mistake. Course-specific consent is obtained _at the time of enrollment_, not before. In fact, we're making modifications to the ecommerce stack to ensure that users are prompted for course-specific consent as part of the redemption process.

So, eligibility in this context isn't a question of "has the user fulfilled every requirement necessary to redeem this offer?", but a question of "has the user done anything to specifically exclude their ability to redeem?"

If that's the goal, I would suggest gating based on an account-level consent requirement, if extant, since account-level requirements are enforced at login, prior to the redemption of any offers.

In contrast, we should _not_ gate on course-specific consent, because such a consent cannot be provided until after the user attempts to take advantage of the offer.

Does that clarify things?